src/pynxtools/dataconverter/validation.py

```diff
@@ -847,6 +859,93 @@ def startswith_with_variations(
         # default
         return (False, 0)
 
+    def check_reserved_suffix(key: str, mapping: MutableMapping[str, Any]) -> bool:
+        """Check if an associated field exists for a key with a reserved suffix."""
+        reserved_suffixes = (
+            "_end",
+            "_increment_set",
+            "_errors",
+            "_indices",
+            "_mask",
+            "_set",
+            "_weights",
+            "_scaling_factor",
+            "_offset",
+        )
+
+        if not key.endswith((reserved_suffixes)):
+            # Ignore this test
+            return False
+
+        for suffix in reserved_suffixes:
+            if key.endswith(suffix) and key.rsplit(suffix, 1)[0] not in mapping:
+                collector.collect_and_log(
+                    key,
+                    ValidationProblem.ReservedSuffixWithoutField,
+                    key.rsplit(suffix, 1)[0],
+                )
+                return False
+        return True
+
+    def check_reserved_prefix(
+        key: str,
+        mapping: MutableMapping[str, Any],
+        nx_type: Literal["group", "field", "attribute"],
+    ) -> bool:
+        """Check if a reserved prefix was used in the correct context."""
+        reserved_prefixes = {
+            "attribute": {
+                "@BLUESKY_": None,  # do not use anywhere
+                "@DECTRIS_": "NXmx",
+                "@IDF_": None,  # do not use anywhere
+                "@NDAttr": None,
+                "@NX_": "all",
+                "@PDBX_": None,  # do not use anywhere
+                "@SAS_": "NXcanSAS",
+                "@SILX_": None,  # do not use anywhere
+            },
+            "field": {
+                "DECTRIS_": "NXmx",
+            },
+        }
+
+        prefixes = reserved_prefixes[nx_type]
```

Comment:
This will give a key error when nx_type == "group" (mabye never happens, but then restrict type of nx_type)